include/sparrow/buffer/dynamic_bitset/dynamic_bitset_base.hpp

```diff
@@ -191,6 +191,13 @@ namespace sparrow
         requires std::ranges::random_access_range<std::remove_pointer_t<B>>
     constexpr auto dynamic_bitset_base<B>::operator[](size_type pos) -> reference
     {
+        if(pos >= size())
+        {
+            throw std::out_of_range(
+                "dynamic_bitset_base::operator[]: index out of range for dynamic_bitset_base of size"
+                + std::to_string(size()) + " at index " + std::to_string(pos)
+            );
+        }
```

Comment:
This is redundant with the assertion and can have a performance cost. operator[] should not systematically check its bounds (we may add an `at` method fir secured access is required).

Reply:
oups my bad, it's a rest of debugging code. I will remove it